### DISCUSSION TITLE
hec-dss-python coordination

### DISCUSSION THREAD

**Discussion by adamkorynta** [CentralValleyModeling/pandss#5] [General]
I received a notification about being added as a subscriber to the project. I don't have access to the Discussions page here, so I don't know where better to post this question:

HEC is developing a python module that allows reading DSS into pandas data frames. Is there any coordination between this project and that one? 

https://github.com/HydrologicEngineeringCenter/hec-dss-python

**Comment by hxiedwr** [comment]
We are re-evaluating if we want to go with the Excel Addin route. Updating Excel Addin when HecLib upgrade is a headache. Management heard of some applications of using python to interact HecLib and likes the idea. Another option is using WRIMS GUI to interact with Excel. Once WRIMS GUI is updated with new HecLib, the GUI can interact with Excel
________________________________
From: Adam Korynta ***@***.***>
Sent: Wednesday, March 20, 2024 2:29:49 PM
To: CentralValleyModeling/pandss ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [CentralValleyModeling/pandss] hec-dss-python coordination (Issue #2)


I received a notification about being added as a subscriber to the project. I don't have access to the Discussions page here, so I don't know where better to post this question:

HEC is developing a python module that allows reading DSS into pandas data frames. Is there any coordination between this project and that one?

https://github.com/HydrologicEngineeringCenter/hec-dss-python

**Comment by adamkorynta** [comment]
yeah there are lots of users wanting python interaction with DSS, that's why I wanted to make you aware of the existing effort in case you wanted to collaborate with them.

**Comment by dwr-zroy** [comment]
In my mind this project provides a common API that sits on top of the various python+DSS projects that already exist. Many of the existing projects have different design goals, and either target only one of DSS 6 or DSS 7, or provide different support for datetime operations, etc. 

You can see that the `DSS` object here utilizes different "engines" to actually do the leg work and interact with DSS files. Right now only `pyhecdss` if fully used in the test suite, but an engine that uses `pydsstools` has been built and partially tested for. Additional engines can be added in the future, and the defaults switched without the API changing.
